tests/sentry/db/test_transactions.py

```diff
@@ -154,3 +159,40 @@ def test_bad_transaction_boundaries(self):
     @django_db_all
     def test_collect_transaction_queries(self):
         super().test_collect_transaction_queries()
+
+
+class FakeControlService:
+    def a(self) -> int:
+        return 1
+
+
+class FakeRegionService:
+    def a(self) -> int:
+        return 2
+
+
+@no_silo_test(stable=True)
+class TestDelegatedByOpenTransaction(TestCase):
+    def test_selects_mode_in_transaction_or_default(self):
+        service: Any = silo_mode_delegation(
+            {
+                SiloMode.CONTROL: lambda: FakeControlService(),
+                SiloMode.REGION: lambda: FakeRegionService(),
+                SiloMode.MONOLITH: lambda: FakeRegionService(),
+            }
+        )
+
+        with override_settings(SILO_MODE=SiloMode.CONTROL):
```

Comment:
`pytest.mark.parametrize` is sometimes useful for these types of matrices, not sure it'd necessarily make things more clear here though